semaphore/Cargo.toml

```diff
@@ -1,49 +0,0 @@
-[package]
```

Comment:
I think this is still useful, we aren't unlikely to want to use this in the feature. There are no problems to keep it in the code base as WIP, and when it gets prioritized to update it.